### PR TITLE
action: prevent globbing with double quotes

### PR DESCRIPTION
**Pull request by ljmf00**
This patch adds double quotes on variables to prevent globbing and prevent evaluation errors such as:

    line 1: [: =: unary operator expected

It also add consistency on bash string equality comparison by using `==`
operator instead of `=`.

Signed-off-by: Luís Ferreira <contact@lsferreira.net>

---

Since this is behaving "good" by accident, please double check if this doesn't break any specific use-case of this action. I recommend changing this to standalone scripts and passing GitHub variables by argument and parse them using something like POSIX getopt. See https://man7.org/linux/man-pages/man1/getopts.1p.html

### CODE REVIEW

**Comment by ljmf00** [comment]
CC @lolgab

**Comment by ale5000-git** [comment]
@ljmf00: "==" is not POSIX compliant and also it won't make any difference in single brackets.
Still to be usefull it should be inside [[ ]] instead of [ ]

**Comment by ljmf00** [comment]
> @ljmf00: "==" is not POSIX compliant and also it won't make any difference in single brackets. Still to be usefull it should be inside [[ ]] instead of [ ]

The usage of non-POSIX complaint features is not a problem here, since the action uses the `bash` shell, and that feature is specified in `bash`. I can't also see that as an argument since the existing code uses bash-specific features after all.

As the commit message suggested, these changes were to improve consistency in one way or another and not about the usefulness of equality/pattern matching in `bash`, as a feature. Arguing about the decision of making it consistent with `==` instead of `=` is that it makes it more readable/less strange among other languages syntax.

**Review by AbsolytnaiaLove — COMMENTED** [review]
#68

**Comment by ljmf00** [comment]
> #68

What do you mean?

File: action.yml [hunk 1]
```diff
@@ -95,11 +95,11 @@ runs:
...
-        echo "COMMIT_SHA=$(if [ ${{ github.event_name }} == "pull_request" ]; then echo "${{ github.event.pull_request.head.sha }}"; else echo "${{ github.sha }}"; fi)" >> $GITHUB_ENV
+        echo "CODACY_BASE_URL_OR_DEFAULT=$(if [ -n "${{ inputs.codacy-api-base-url }}" ]; then echo "${{ inputs.codacy-api-base-url }}"; else echo "https://api.codacy.com"; fi)" >> "$GITHUB_ENV"
+        echo "OWNER_NAME=$(echo "$GITHUB_REPOSITORY" | cut -d '/' -f 1)" >> "$GITHUB_ENV"
+        echo "REPOSITORY_NAME=$(echo "$GITHUB_REPOSITORY" | cut -d '/' -f 2)" >> "$GITHUB_ENV"
+        echo "ORGANIZATION_PROVIDER=$(if [ "$GITHUB_SERVER_URL" == "https://github.com" ]; then echo "gh"; else echo "ghe"; fi)" >> "$GITHUB_ENV"
+        echo "COMMIT_SHA=$(if [ "${{ github.event_name }}" == "pull_request" ]; then echo "${{ github.event.pull_request.head.sha }}"; else echo "${{ github.sha }}"; fi)" >> "$GITHUB_ENV"
         if [ -n "${{ inputs.skip-container-engine-check }}" ]; then
           echo "SKIP_CONTAINER_ENGINE_CHECK=${{ inputs.skip-container-engine-check }}" >> $GITHUB_ENV
```
Comment (mrfyda) [inline action.yml]:
```suggestion
          echo "SKIP_CONTAINER_ENGINE_CHECK=${{ inputs.skip-container-engine-check }}" >> "$GITHUB_ENV"
```

**Review by mrfyda — APPROVED** [review]

**Pull request merged**